tests/bluetooth/ctrl_isoal/src/sub_sets/isoal_test_rx.c

```diff
@@ -925,14 +925,18 @@ ZTEST(test_rx_unframed, test_rx_time_wrapping)
 
 	/* Maximum negative difference from 0 */
 	time_now = 0;
-	time_diff = (time_wrapping_point == UINT32_MAX ? INT32_MIN : -ISOAL_TIME_WRAPPING_POINT_US);
+	time_diff = (time_wrapping_point == UINT32_MAX) ?
+		    INT32_MIN :
+		    -((int64_t)ISOAL_TIME_WRAPPING_POINT_US);
```

Comment:
`time_diff` is `int32_t`, so it doesn't make sense to cast to `int64_t`

Reply:
I think it looks alright. With this timer I would expect `time_wrapping_point == UINT32_MAX` to be true so the value actually assigned should be `INT32_MIN`.

Reply:
> Compiler did not like when ISOAL_TIME_WRAPPING_POINT_US is 32-bit unsigned int, as signed values need an extra bit to represent them in 2's compliment, hence have to typecast to 64-bit, then the compiler is ok to assign it to int32_t.

What is the compiler error/warning? 

Depending on what parameters is provided to e.g. GCC, it would also give an error/warning if you assigned a `int64_t` to a `int32_t`, so I would suggest to do it properly than just replacing one compiler error/warning with a future error/warning. 

One easy solution would be to simply have `time_diff` as `int64_t` and then if the result actually fits in a `int32_t`, then we can cast the result.

Reply:
The compile error was:

> /home/jack/workspace/zephyrproject/zephyr/tests/bluetooth/ctrl_isoal/src/sub_sets/isoal_test_rx.c: In function 'test_rx_unframed_test_rx_time_wrapping':
/home/jack/workspace/zephyrproject/zephyr/tests/bluetooth/ctrl_isoal/src/sub_sets/isoal_test_rx.c:928:21: warning: overflow in conversion from 'long long unsigned int' to 'int32_t' {aka 'int'} changes value from '18446744073197551647' to '-511999969' [-Woverflow]
  928 |         time_diff = (time_wrapping_point == UINT32_MAX ? INT32_MIN : -ISOAL_TIME_WRAPPING_POINT_US);
      |                     ^
/home/jack/workspace/zephyrproject/zephyr/tests/bluetooth/ctrl_isoal/src/sub_sets/isoal_test_rx.c:935:21: warning: overflow in conversion from 'long long unsigned int' to 'int32_t' {aka 'int'} changes value from '18446744073197551647' to '-511999969' [-Woverflow]
  935 |         time_diff = (time_wrapping_point == UINT32_MAX ? INT32_MIN : -ISOAL_TIME_WRAPPING_POINT_US);
      |                     ^